File: app/Models/Tournament.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Tournament extends Model
{
    protected $fillable = ['name', 'start_date', 'end_date', 'location'];

    public function teams(): BelongsToMany
    {
        return $this->belongsToMany(Team::class, 'tournament_teams', 'tournament_id', 'team_id');
    }

    public function games(): BelongsToMany
    {
        return $this->belongsToMany(Game::class, 'tournament_games', 'tournament_id', 'game_id');
    }
}

File: app/db/models/Tournament.py
from sqlalchemy import Column, Integer, String, Date
from sqlalchemy.orm import relationship
from app.db import Base

class Tournament(Base):
    __tablename__ = "tournaments"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    start_date = Column(Date, nullable=False)
    end_date = Column(Date, nullable=True)
    location = Column(String, nullable=True)
    logo_url = Column(String, nullable=True)

    # Association object collections
    tournament_teams = relationship("TournamentTeams", back_populates="tournament", cascade="all, delete-orphan")
    tournament_players = relationship("TournamentPlayers", back_populates="tournament", cascade="all, delete-orphan")
    tournament_games = relationship("TournamentGames", back_populates="tournament", cascade="all, delete-orphan")

    # Convenience view-only collections to access related entities directly
    teams = relationship("Team", secondary="tournament_teams", viewonly=True)
    players = relationship("Player", secondary="tournament_players", viewonly=True)
    games = relationship("Game", secondary="tournament_games", viewonly=True)